src/main.rs

```diff
@@ -27,22 +27,34 @@ fn main() {
         .unwrap()
         .block_on(async move {
             let mut config: Config = match args.config.as_str() {
-                STDIN_FILE_DESIGNATOR => { // read from stdin
+                STDIN_FILE_DESIGNATOR => {
+                    // read from stdin
                     let mut config_str = String::new();
 
-                    let size = std::io::stdin().read_to_string(&mut config_str)
-                    .or_error(|| format!("Configuration file could not be read from stdin"))?;
-
-                    if size == 0 {                        
-                        return Err(i3status_rs::errors::Error { kind: ErrorKind::Config, message: None, cause: None, block: None });
+                    let size = std::io::stdin()
```

Comment:
You probably need to use the tokio version of stdin. https://docs.rs/tokio/latest/tokio/io/fn.stdin.html